react/src/components/home/Home.jsx

```diff
@@ -25,55 +25,185 @@ function Home() {
+    firstFilteredResults: [],
+    secondFilteredResults: [],
+  });
+
+  const [query, setQuery] = useState("");
+
+  const [locationFilter, setLocationFilter] = useState("");
+
+  const handleSearch = (e) => {
+    setQuery(e.target.value);
+  };
+
+  const handleLocation = (e) => {
+    setLocationFilter(e.target.value);
+  };
+
+  const onSearchRequested = (e) => {
+    e.preventDefault();
+
+    resultsData.firstFilteredResults = filterByValue(results, query);
+
+    if (locationFilter) {
+      resultsData.secondFilteredResults = filterByLocation(
+        resultsData.firstFilteredResults,
+        locationFilter
+      );
+    } else {
+      resultsData.secondFilteredResults = resultsData.firstFilteredResults;
+    }
+
+    setResultsData((prevState) => {
+      let newResults = resultsData.secondFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onLocationRequested = (e) => {
+    e.preventDefault();
+
+    resultsData.firstFilteredResults = filterByLocation(
+      results,
+      locationFilter
+    );
+
+    if (query) {
+      resultsData.secondFilteredResults = filterByValue(
+        resultsData.firstFilteredResults,
+        query
+      );
+    } else {
+      resultsData.secondFilteredResults = resultsData.firstFilteredResults;
+    }
+
+    setResultsData((prevState) => {
+      let newResults = resultsData.secondFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onResetQueryFilter = (e) => {
+    e.preventDefault();
+
+    setQuery("");
+    clearQueryField();
+
+    if (locationFilter) {
+      resultsData.firstFilteredResults = filterByLocation(
+        results,
+        locationFilter
+      );
+    } else if (!locationFilter) {
+      resultsData.firstFilteredResults = results;
+    }
+    setResultsData((prevState) => {
+      let newResults = resultsData.firstFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onResetLocationFilter = (e) => {
+    e.preventDefault();
+
+    setLocationFilter("");
+    clearLocationField();
+
+    if (query) {
+      resultsData.firstFilteredResults = filterByValue(results, query);
+    } else if (!query) {
+      resultsData.firstFilteredResults = results;
+    }
+    setResultsData((prevState) => {
+      let newResults = resultsData.firstFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  function filterByValue(array, string) {
+    console.log(string);
+
+    return array.filter((o) =>
+      Object.keys(o).some((k) =>
+        String(o[k]).toLowerCase().includes(string.toLowerCase())
```

Comment:
I'd like to see you take this whole return statement and refactor it, Sabio-style. It looks to me like code straight from Stack Overflow, so I challenge you to refactor it to show your understanding of it. Could you please break this out into functions please (not inline functions)